File: backend/app/modules/gestion_usuarios/schemas/technician.py
from datetime import datetime

from pydantic import Field

from app.schemas.common import ORMBaseModel


class TechnicianCreateRequest(ORMBaseModel):
    name: str = Field(min_length=1, max_length=100)
    phone: str | None = Field(default=None, max_length=20)
    specialty: str | None = Field(default=None, max_length=100)
    is_available: bool = True


class TechnicianUpdateRequest(ORMBaseModel):
    name: str = Field(min_length=1, max_length=100)
    phone: str | None = Field(default=None, max_length=20)
    specialty: str | None = Field(default=None, max_length=100)
    is_available: bool


class TechnicianAvailabilityUpdateRequest(ORMBaseModel):
    is_available: bool


class TechnicianResponse(ORMBaseModel):
    id_technician: int
    id_workshop: int
    name: str
    phone: str | None = None
    specialty: str | None = None
    is_available: bool
    created_at: datetime

File: frontend/src/app/models/technician.ts
export interface TechnicianResponse {
  id_technician: number;
  id_workshop: number;
  name: string;
  phone: string | null;
  specialty: string | null;
  is_available: boolean;
  created_at: string;
}

export interface TechnicianCreateRequest {
  name: string;
  phone?: string | null;
  specialty?: string | null;
  is_available?: boolean;
}

export interface TechnicianUpdateRequest {
  name: string;
  phone?: string | null;
  specialty?: string | null;
}

export interface TechnicianAvailabilityUpdateRequest {
  is_available: boolean;
}
